### PR TITLE
Improve message that opening a folder is required for some commands

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Improves message for #532.
Also combines the code and correctly throws rather than just showing messages.

### CODE REVIEW

File: commands/build-image.ts [hunk 1]
```diff
@@ -56,25 +57,7 @@ export async function buildImage(actionContext: IActionContext, dockerFileUri: v
...
-
-    if (!rootFolder) {
-        if (!vscode.workspace.workspaceFolders) {
-            vscode.window.showErrorMessage('Docker files can only be built if VS Code is opened on a folder.');
-        } else {
-            vscode.window.showErrorMessage('Docker files can only be built if a workspace folder is picked in VS Code.');
-        }
-        return;
-    }
+    let rootFolder: vscode.WorkspaceFolder = await quickPickWorkspaceFolder('To build Docker files you must first open a folder or workspace in VS Code.');
```
Comment (PrashanthCorp) [inline commands/build-image.ts]: I see `you must first open a folder or workspace in VS Code.` being repeated in all 3 calls. Possibly add that to the function itself?

Comment (StephenWeatherford) [reply commands/build-image.ts]: I'd rather not. Doing string concatenation of parts of sentences causes problems if we want to localize later.

Comment (PrashanthCorp) [reply commands/build-image.ts]: Interesting. I thought the loc could happen in the function, at the point we throw the error. 
I don't feel strongly about this, I was just wondering if there was a way to force consistency in the messages in the future. These errors would stand out to the user more than to they would to the dev.

Comment (StephenWeatherford) [reply commands/build-image.ts]: I assumed you mean something like:
   `To ${message} you need to first open...`

That tends to be problematic.  I agree this is non-optimal, but I haven't found a better way, except to restrict the message to be more generic. I think this is good enough, though.  Thx.

**Pull request merged**